app/helpers/react_on_rails_helper.rb

```diff
@@ -102,11 +104,13 @@ def react_component(component_name, raw_options = {})
     # The reason is that React is smart about not doing extra work if the server rendering did its job.
 
     component_specification_tag =
-      content_tag(:div,
-                  "",
+      content_tag(:script,
                   class: "js-react-on-rails-component",
-                  style: options.style,
-                  data: options.data)
+                  style: nil,
+                  data: options.data) do
+        props = Yajl.dump(options.props.is_a?(String) ? JSON.parse(options.props) : options.props)
+        "var #{options.dom_id.tr('-', '_')} = #{props};".html_safe
```

Comment:
It also looks like JSON <script> tag also solves the Turbolinks issue, because there is no need to execute any <script> tag, you just have to replace `el.getAttribute()` with (IIRC) `.textContent`.  Also, no `eval()`!